### ISSUE TITLE
Streamline the creation of utilities

### ISSUE THREAD

**Issue by luchian** [pixeltrove/spark-css#2]
Editing, adding or removing utility classes manually is cumbersome and error-prone. Provide a mixin that outputs the necessary utilities when presented with the appropriate input.

**Comment by luchian** [comment]
Generate utility classes using the property name, a set of property values, and query names as inputs.

A property can take on multiple values so these are best passed in the form of a variable.
```
$display-values: block, flex, none;
@include utility(display, $display-values);
```

Responsive classes could be generated for all designated property values. However, fine-tuning the output would significantly lower file-size. Use maps to establish what responsive classes should be created.
```
$display-values: (
  standard: (block, flex),
  palm: none
);
@include utility(display, $display-values);
```
This should generate a `u-display-none@palm` class within the appropriate media query.

**Comment by luchian** [comment]
Some property values will require custom names. These should be defined via map keys.
```
$margin-right-values: (
  regular: 1rem,
  small: 0.5rem
);
```
Responsive classes should also be allowed to incorporate custom names.
```
$margin-right-values: (
  all: (
    regular: 1rem,
    small: 0.5rem
  )
);
```
This should create all possible non-responsive and responsive classes for the specified values.

**Comment by luchian** [comment]
Some property values will require custom names. These should be defined via map keys.
```
$margin-right-values: (
  regular: 1rem,
  small: 0.5rem
);
$margin-bottom-values: (
  all: (
    regular: 1rem,
    small: 0.5rem
  )
);
```

**Comment by luchian** [comment]
Names such as `1\/2` cannot be used as map keys because the escape character is automatically escaped resulting in `1\\/2` which leads to invalid class names. Such designations have to use interpolation `1#{\/}2`.

**Comment by luchian** [comment]
Addressed in v0.6.0.
